torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1445,16 +1442,11 @@ def _shard_parameters(self) -> None:
             local_shard, numel_padded = FlatParamHandle._get_shard(p, self.rank, self.world_size)
             p.set_(local_shard)  # type: ignore[call-overload]
             handle.init_shard_metadata(local_shard.numel(), numel_padded, self.rank)
-            self.numel_padded_per_param.append(numel_padded)
 
             # Free storage that contains the original full data.
             if orig_storage.size() > 0:
                 orig_storage.resize_(0)  # type: ignore[attr-defined]
 
-        assert len(self.numel_padded_per_param) == len(
```

Comment:
Does losing this assert make it more likely we might hit bugs/issues?

Reply:
It should be very safe to remove since there is an explicit loop over the handles, where there is one parameter per handle, and one append per handle. I think the assert was legacy from the initialize upstream from Fairscale and may have served a purpose at that time.